astropy/utils/data.py

```diff
@@ -1008,18 +1009,22 @@ def download_file(remote_url, cache=False, show_progress=True, timeout=None):
             _dataurls_to_alias[conf.dataurl] = [conf.dataurl]
     try:
         if cache:
-            # We don't need to acquire the lock here, since we are only reading
-            with shelve.open(urlmapfn) as url2hash:
-                if url_key in url2hash:
-                    return url2hash[url_key]
-                # If there is a cached copy from mirror, use it.
-                else:
-                    for cur_url in _dataurls_to_alias.get(conf.dataurl, []):
-                        if url_key.startswith(cur_url):
-                            url_mirror = url_key.replace(cur_url,
-                                                         conf.dataurl_mirror)
-                            if url_mirror in url2hash:
-                                return url2hash[url_mirror]
+            try:
```

Comment:
If you just want to suppress an exception you can also use [`contextlib.suppress`](https://docs.python.org/3/library/contextlib.html#contextlib.suppress)